test/jdk/javax/sound/midi/BulkSoundBank/BulkSoundBank.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.sound.midi.InvalidMidiDataException;
+import javax.sound.midi.MidiSystem;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+/**
+ * @test
+ * @bug 8350813
+ * @summary Rendering of bulky sound bank from MIDI sequence can cause OutOfMemoryError.
+ * @run main/othervm -Xmx1g BulkSoundBank
```

Comment:
The test can still throw OOME with `@run main/othervm -Xmx2g BulkSoundBank` after the fix. (Tested on Ubuntu 22.04)

So it looks like the condition when we throw the `InvalidMidiDataException` should be tuned.

Reply:
We can not predict OOME reliably - if size of the rendered data is 99% of the remaining heap and the file i/o or background task takes more than 2% of it - the memory will eventually end, there always will be corner cases no matter what treshold we choose. In the end rendering such a huge bank will take minutes and background task might consume more memory at the time. The code here stops OOME when we know for sure we will not have enough resources and lowering the amount might stop the legitimate use case from working.